op-challenger/fault/cannon/updater.go

```diff
@@ -28,8 +29,27 @@ type cannonUpdater struct {
 	preimageOracleAddr common.Address
 }
 
-// NewOracleUpdater returns a new updater.
+// NewOracleUpdater returns a new updater. The pre-image oracle address is loaded from the fault dispute game.
 func NewOracleUpdater(
+	ctx context.Context,
+	logger log.Logger,
+	txMgr txmgr.TxManager,
+	fdgAddr common.Address,
+	client bind.ContractCaller,
+) (*cannonUpdater, error) {
+	gameCaller, err := bindings.NewFaultDisputeGameCaller(fdgAddr, client)
+	if err != nil {
+		return nil, fmt.Errorf("create caller for game %v: %w", fdgAddr, err)
+	}
+	oracleAddr, err := gameCaller.BLOCKORACLE(&bind.CallOpts{Context: ctx})
```

Comment:
This is the `BlockOracle` contract, not the `PreimageOracle` contract - the `PreimageOracle` address can be fetched from the `MIPS` VM.

Reply:
oh good spot. I'll fix it up.